### PR TITLE
[LMLayer] Documents the methods of WorkerInternalModel

### PR DESCRIPTION
**Pull request by jahorton**
As noted in feedback for #1869, we're currently missing documentation for some of the core functions.  This PR seeks to address those concerns... though I'm afraid it might be a bit wordy.

### CODE REVIEW

**Comment by keyman-server** [comment]
[OSK renders for Windows](https://build.palaso.org/repository/download/KeymanDesktop_TestPrRenderOnScreenKeyboards/158029:id/kvk/index.html)

File: common/predictive-text/worker/worker-interfaces.ts [hunk 1]
```diff
@@ -135,8 +135,54 @@ interface LMLayerWorkerState {
...
+  /**
+   * Processes `config` messages, configuring the newly-loaded model based on the host
+   * platform's capability restrictions.
+   * 
+   * This allows the model to configure its suggestions according to what the platform
+   * allows the host to actually perform - for example, if post-caret deletions are not
+   * supported, no suggestions requiring this feature should be produced by the model.
+   * 
+   * Returns a `Configuration` object detailing the capabilities the model plans to
+   * actually utilize, which must be more restrictive than those indicated within
```
Comment (eddieantonio) [inline common/predictive-text/worker/worker-interfaces.ts]: Small nitpick: "must be **more** restrictive" than the capabilities? Isn't this more like "can match the capabilities OR be more restrictive"?

File: common/predictive-text/worker/worker-interfaces.ts [hunk 2]
```diff
@@ -135,8 +135,54 @@ interface LMLayerWorkerState {
...
+   * @param capabilities 
+   */
   configure(capabilities: Capabilities): Configuration;
+
+  /**
+   * Generates predictive suggestions corresponding to the state of context resulting from
+   * applying a potential transform to the current context state.  This transform may correspond 
+   * to a 'correction' of a recent keystroke rather than one actually received.
+   * 
+   * This method should NOT attempt to perform any form of correction; this is modeled within a
```
Comment (eddieantonio) [inline common/predictive-text/worker/worker-interfaces.ts]: 👍

File: common/predictive-text/worker/worker-interfaces.ts [hunk 3]
```diff
@@ -135,8 +135,54 @@ interface LMLayerWorkerState {
...
+   * 
+   * Returns a `Configuration` object detailing the capabilities the model plans to
+   * actually utilize, which must be more restrictive than those indicated within
+   * the provided `Capabilities` object.
+   * @param capabilities 
+   */
   configure(capabilities: Capabilities): Configuration;
+
+  /**
+   * Generates predictive suggestions corresponding to the state of context resulting from
```
Comment (eddieantonio) [inline common/predictive-text/worker/worker-interfaces.ts]: The brief description here, while correct, is quite wordy, and a bit confusing. Could you please try simplifying the wording here, leaving the details to the following paragraphs?

File: common/predictive-text/worker/worker-interfaces.ts [hunk 4]
```diff
@@ -135,8 +135,54 @@ interface LMLayerWorkerState {
...
+   * @param context A depiction of the context to which `transform` is applied.
+   * @returns A probability distribution (`Distribution<Suggestion>`) on the resulting `Suggestion` 
+   * space for use in determining the most optimal overall suggestions.
+   */
   predict(transform: Transform, context: Context): Distribution<Suggestion>;
+
+  /**
+   * Performs a wordbreak operation given the current context state, returning whatever word
+   * or word fragment exists that starts before the caret but after the most recent whitespace
+   * preceedin the caret.
```
Comment (eddieantonio) [inline common/predictive-text/worker/worker-interfaces.ts]:
```suggestion
   * preceding the caret.
```

File: common/predictive-text/worker/worker-interfaces.ts [hunk 5]
```diff
@@ -135,8 +135,54 @@ interface LMLayerWorkerState {
...
+   *
```
Comment (eddieantonio) [inline common/predictive-text/worker/worker-interfaces.ts]: A thing missing here is what happens when the `wordbreak()` is called directly after whitespace in the left context? From my understanding of the clients of `.wordbreak()`, the appropriate return value is an empty string, but that is not documented in this interface. Could you please elaborate on this?

File: common/predictive-text/worker/worker-interfaces.ts [hunk 6]
```diff
@@ -135,8 +135,54 @@ interface LMLayerWorkerState {
...
+   * @param context A depiction of the context to which `transform` is applied.
+   * @returns A probability distribution (`Distribution<Suggestion>`) on the resulting `Suggestion` 
+   * space for use in determining the most optimal overall suggestions.
+   */
   predict(transform: Transform, context: Context): Distribution<Suggestion>;
+
+  /**
+   * Performs a wordbreak operation given the current context state, returning whatever word
+   * or word fragment exists that starts before the caret but after the most recent whitespace
+   * preceding the caret.  If no such text exists, the the empty string is returned.
```
Comment (eddieantonio) [inline common/predictive-text/worker/worker-interfaces.ts]:
```suggestion
   * preceding the caret.  If no such text exists, the empty string is returned.
```
